### DISCUSSION TITLE
24.06.03 Progress Meeting

### DISCUSSION THREAD

**Discussion by ssutjipto** [Woolfrey/software_robot_library#123] [Meetings]
#### Chair:
@ssutjipto 

#### Attendees:
@Woolfrey 
@ssutjipto 
_____

## Announcements
Shoudong is coming to Leeds end of July, Sheila is contact point in Australia

## Agenda
_Note: Last week's actions become this week's agenda_

| Item | Responsible | Issue | Outcome |
|-------|------------------|----------|--------------|
| Finish the test script for velocity level joint trajectory tracking | @Woolfrey | #116 | Yes, but just need to test a 6DOF robot  |
| Continue on action server | @Woolfrey | [Link](https://github.com/Woolfrey/server_serial_link/issues/1)  | Continuing work on this |
| Type up math for joint level feedback in discussion below | @Woolfrey | #115 | Yes, see comments. |
| Continue working on dynamic control class | @ssutjipto | #117 | Still working on it. |

## Additional Discussion
| Topic | Raised By | Issue | Outcome |
|---------|---------------|---------|--------------|
| Make a habit of adding damping torque | @Woolfrey | N/A | N/A | 
| Is the control limit calculation correct | @ssutjipto | N/A | Yes, the equations are correct in the SerialKinematicControl class. |
| Don't invert matrix directly, if it can be avoided | @Woolfrey | N/A | See comments from this discussion. |
| Test robustness of the singularity avoidance strategy | @Woolfrey | N/A | In future, will need to test this under different conditions. |

## Actions
| Action | Responsible | Tag |
|----------|------------------|-------|
| Test 6DOF Cartesian control | @Woolfrey | #122 |
| Continue working on dynamic control class | @ssutjipto | #117 |
| Set default gains as function of control frequency | @Woolfrey @ssutjipto  | To be created |
| Test default gains | @Woolfrey @ssutjipto | To be created |

**Comment by Woolfrey** [comment]
Learn these: https://web.archive.org/web/20240309190945/https://eigen.tuxfamily.org/dox/group__DenseDecompositionBenchmark.html

For example,
```
Eigen::VectorXd qddot = M.ldlt().solve(tau);
```

**Comment by Woolfrey** [comment]
[Notes_on_Constrained_Dynamics.pdf](https://github.com/user-attachments/files/15531552/Notes_on_Constrained_Dynamics.pdf)
